vlib/vweb/vweb.v

```
@@ -315,16 +315,12 @@ pub fn (mut ctx Context) not_found() Result {
 
 // TODO - test
 // Sets a cookie
-pub fn (mut ctx Context) set_cookie(cookie http.Cookie) {
-	mut cookie_data := []string{}
-	mut secure := if cookie.secure { 'Secure;' } else { '' }
-	secure += if cookie.http_only { ' HttpOnly' } else { ' ' }
-	cookie_data << secure
-	if cookie.expires.unix > 0 {
-		cookie_data << 'expires=${cookie.expires.utc_string()}'
+pub fn (mut ctx Context) set_cookie(cookie http.Cookie) ! {
+	cookie_raw := cookie.str()
+	if cookie_raw == '' {
+		return error('error setting cookie: name of cookie is invalid')
```

ah I see you're doing it in vweb

I don't think it should return a Result, just do nothing if it's empty

---

Was doubting about retunring an error, but saw a comment from JalonSolov on the issue that `set_cookie` should return an error. I also tend to lean more towards not returning an error.

---

`set_cookie` should return an error if you try to set a cookie that doesn't match the specs.

---

Or I guess it's not a change any longer, since this PR was already merged.  <sigh>

---

@JalonSolov neither Go, nor ASP.NET MVC, nor Flask return an error.

```
SetCookie adds a Set-Cookie header to the provided ResponseWriter's headers. The provided cookie must have a valid Name. Invalid cookies may be silently dropped.
```

https://learn.microsoft.com/en-us/dotnet/api/system.web.httpcookiecollection.add?view=netframework-4.8.1#system-web-httpcookiecollection-add(system-web-httpcookie)

---

We want to be consistent with the industry. Result also makes handling a very common operation more verbose.

---

If "consistent with the industry" equates to "bug for bug compatible", then sure.  I was just hoping V could do better.

---

@spytheman what do you think?